### PR TITLE
refactor(flags): rename CheckFlag to TypecheckMode

### PR DESCRIPTION
**Pull request by bartlomieju**
Working on https://github.com/denoland/deno/pull/14072 it became confusing to have `CheckFlag` and `CheckFlags`. I believe `TypecheckMode` is a better name than `CheckFlag`. 

Also renamed `Flags.check` to `Flags.typecheck_mode` and `CheckOptions.check` to `CheckOptions.typecheck_mode`.

### CODE REVIEW

File: cli/flags.rs [hunk 1]
```diff
@@ -215,7 +215,7 @@ impl Default for DenoSubcommand {
 }
 
 #[derive(Debug, Clone, PartialEq)]
-pub enum CheckFlag {
+pub enum TypecheckMode {
```
Comment (dsherret) [inline cli/flags.rs:218]: Maybe `TypeCheckMode`? I'm not sure if it's considered one or two words.

**Review by dsherret — APPROVED** [review]
LGTM

**Comment by bartlomieju** [comment]
@kitsonk are you fine with this change?

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**